java/dependencyManagement/build.gradle.kts

```diff
@@ -10,10 +10,14 @@ data class DependencySet(val group: String, val version: String, val modules: Li
 
 val DEPENDENCY_BOMS = listOf(
     "io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom-alpha:1.0.1-alpha",
+    "org.apache.logging.log4j:log4j-bom:2.14.0",
     "software.amazon.awssdk:bom:2.15.67"
 )
 
 val DEPENDENCIES = listOf(
+    "com.amazonaws:aws-lambda-java-core:1.2.1",
+    "com.amazonaws:aws-lambda-java-events:3.7.0",
+    "com.squareup.okhttp3:okhttp:4.9.0",
     "io.opentelemetry.javaagent:opentelemetry-javaagent:1.0.1"
```

Comment:
If we use OTel javaagent but not aws OTel javaagent, need to inject aws id generator and propagator in sample app?

Reply:
For this repo we're not using x-ray since it's not the aws repo

Reply:
we may want xray backend for integration test in CI/CD.

Reply:
Yeah I think we'd want to have those in an aws-observability repo, xray is vendor-specific so we normally wouldn't have tracing vendor-specific tests here right?

Reply:
It is hard to make Lambda repo to be vendor agnostic since Lambda is aws, take the example, we will use ADOT collector in Collector extension because we have no better choice. If the config is open it is ok to set aws as default. 

Hence the CI/CD, Lambda layer will also be in this repo. The aws-observability/aws-otel-lambda repo will be deprecated.

Reply:
Yeah that's cloud vendor - naturally it will have AWS-specific logic. But not tracing-vendor (X-Ray) specific code with a few exceptions for simple components like propagation. I'm not aware of significant tracing-vendor-specific work being OK in open-telemetry org, even for this repo, since it's the job of the vendor. @alolita what do you think?

Reply:
Some open questions:
1. Do we want to release public layers in this repo or in downstream repo?
2. Do we want to have a vendor agnostic Collector in Lambda extension or not?
3. Do we want to have a vendor agnostic SDK Lambda layer? 
4. Do we want to have vendor agnostic sample apps in this repo?

Reply:
> Do we want to release public layers in this repo or not(in downstream repo)?

Yeah - I think we can release public layers for what I've configured here. They wouldn't work with x-ray but should for every other vendor if they're OpenTelemetry-compliant. This is a problem for x-ray to fix, not this repo though and in the meantime anything published from here should reflect the core OpenTelemetry spec.

> Do we want to have a vendor agnostic Collector in Lambda extension or not? // in other word, can we use AWS distro collector?

I figured we would publish 1-2 layers of `opentelemetry-collector` and/or `opentelemetry-collector-contrib`. Customers will generally need to build their extension anyways but the reference point should be artifacts under control of open-telemetry.

> The sample app needs aws account to deploy to aws Lambda, it naturally exports data to AWS backend like Cloudwatch and XRay for user's convenience.

There will be some automatic reporting regardless of user configuration. But the configuration we present needs to be neutral - this is open-telemetry, so for example, our sample apps cannot set `tracing_mode`, that is an x-ray feature and loses the vendor-neutrality of this project.

Reply:
Going to go ahead and merge this PR since it's mostly ready, we can decide the future of what artifacts to document, or this repo itself, in followups